sonarts-core/src/tools/new-rule.ts

```diff
@@ -0,0 +1,271 @@
+    showHelp();
+    throw err;
+  }
+}
+
+function getDescription(rspecRuleFolder: string, rspecId: string) {
+  try {
+    let replaceDict: any = {};
+    replaceDict["<em>"] = "";
+    replaceDict["</em>"] = "";
+    replaceDict["<p>"] = "";
+    replaceDict["</p>"] = `\n`;
+    replaceDict["<code>"] = "`";
+    replaceDict["</code>"] = "`";
+    replaceDict["<pre>"] = "```typescript";
+    replaceDict["</pre>"] = "```";
+    replaceDict["<h2>"] = "## ";
+    replaceDict["</h2>"] = `\n`;
+
+    const fileText = fs.readFileSync(`${rspecRuleFolder}/${rspecId}.html`).toString();
+    return replace(fileText, replaceDict);
+  } catch (err) {
+    console.error("could not find metadata for the rule. Have you run the rspec-api?");
+    showHelp();
+    throw err;
+  }
+}
+
+function updateReadme(rootFolder: string, ruleTitle: string, ruleNameDash: string) {
+  const readmePath = `${rootFolder}/README.md`;
+  let readme = fs.readFileSync(readmePath).toString();
+
+  const lines = readme.split("\n");
+
+  let result = [];
+
+  let resultsBlock1 = [];
+  let resultsBlock2 = [];
+  let resultsBlock3 = [];
+
+  let ruleValue = `* ${ruleTitle} ([\`${ruleNameDash}\`])`;
+  resultsBlock1.push(ruleValue);
+  let ruleKey = `[\`${ruleNameDash}\`]: ./sonarts-core/docs/rules/${ruleNameDash}.md`;
+  resultsBlock2.push(ruleKey);
+
+  // 0: start
+  // 1: ## Rules
+  // 2: newlines
+  // 3: rules
+  // 4: newlines
+  // 5: rules 2
+  // 6: newline
+  // 7: after
+
+  let state = 0;
+
+  for (var i = 0; i < lines.length; i++) {
+    let line = lines[i];
+
+    switch (state) {
+      case 0:
+        if (line.trim() == "## Rules") {
+          state = 1;
+        }
+        result.push(line);
+        break;
+
+      case 1:
+      case 2:
+        if (line.length > 0) {
+          state++;
+          resultsBlock1.push(line);
+        } else {
+          result.push(line);
+        }
+
+        break;
+
+      case 3:
+        if (line.length == 0) {
+          state++;
+        } else {
+          resultsBlock1.push(line);
+        }
+        break;
+
+      case 4:
+        if (line.length == 0) {
+          state++;
+        } else {
+          resultsBlock2.push(line);
+        }
+        break;
+      case 5:
+        resultsBlock3.push(line);
+        break;
+    }
+  }
+
+  if (resultsBlock1.length !== resultsBlock2.length) {
+    console.log("resultsBlock1");
+    console.log(resultsBlock1);
+
+    console.log("resultsBlock2");
+    console.log(resultsBlock2);
+
+    throw new Error("could not parse README.md!");
+  }
+
+  var dict2: any = {};
+  for (let i = 0; i < resultsBlock1.length; i++) {
+    dict2[resultsBlock2[i]] = resultsBlock1[i];
+  }
+
+  let keys = Object.keys(dict2).sort();
+  let values = new Array();
+
+  for (let i = 0; i < keys.length; i++) {
+    values.push(dict2[keys[i]]);
+  }
+
+  for (let i = 0; i < values.length; i++) {

```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Expected a 'for-of' loop instead of a 'for' loop with this simple iteration ([typescript:S4138](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4138))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=michalb%2Fnew-rules-script&issues=AWC8Oukl_Q-0dz4JAzUU&open=AWC8Oukl_Q-0dz4JAzUU)

[ISSUE_KEY]: AWC8Oukl_Q-0dz4JAzUU